internal/armada/submit/validation/submit_request_test.go

```diff
@@ -817,6 +818,30 @@ func TestValidateResources(t *testing.T) {
 			}),
 			expectSuccess: false,
 		},
+		"Requests and limits different, passes MaxResourceOversubscriptionByResourceRequest": {
```

Comment:
Add tests for the other cases
 - limit less than request 
 - request and limit have different resources specified